### PR TITLE
gcp - add  cloud storage json  tests

### PR DESCRIPTION
**Pull request by Dkhanas**
Added tests for buckets resources in Cloud Storage Json service

### CODE REVIEW

File: tools/c7n_gcp/tests/test_storage.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    def test_bucket_query(self):
+        project_id = 'test-project-232910'
+        factory = self.replay_flight_data('bucket-query', project_id)
+        p = self.load_policy(
+            {'name': 'all-buckets',
+             'resource': 'gcp.bucket'},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_storage.py]: We need to be more explicit about the results we expect to be returned. What I mean by this is if we expect 7 results we should assertEqual on the specific list of 7 results. 

So if you expect a policy run to return 7 results with the names "location1, location2, location3, location4, location5, location6, location7", then verify that is what you get back in your test.

**Comment by jondavidi** [comment]
- why is there a binary file committed in this PR?
- I think this is an instance where the recorded test data could be reduced. In another PR review, from my understanding of his comment, Sonny @thisisshi seemed to indicate that `get` tests could use the same test data as the query test

**Comment by Dkhanas** [comment]
> * why is there a binary file committed in this PR?
> * I think this is an instance where the recorded test data could be reduced. In another PR review, from my understanding of his comment, Sonny @thisisshi seemed to indicate that `get` tests could use the same test data as the query test

1. Without that file, tests does not work.
2. We can't merge test files, because that is different requests

**Comment by kapilt** [comment]
the discovery docs from gcp are cached as binary files for the test suite to run offline and without credentials, ie. its a normal part of gcp provider testing.

**Comment by mediapills** [comment]
Created pull request in upstream [#3782](https://github.com/cloud-custodian/cloud-custodian/pull/3782).

**Pull request closed without merging**